src/Uno.UI/UI/Xaml/Controls/MediaPlayer/MediaPlayerElement.cs

```diff
@@ -0,0 +1,270 @@
+		{
+			Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () =>
+			{
+				if (PosterSource != null && _posterImage != null)
+				{
+					_posterImage.Visibility = Visibility.Visible;
+					_mediaPlayerPresenter.Opacity = 0;
+				}
+			});
+		}
+
+		private void OnMediaOpened(Windows.Media.Playback.MediaPlayer session, object args)
+		{
+			Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () =>
+			{
+				if (_posterImage != null)
+				{
+					_posterImage.Visibility = Visibility.Collapsed;
+					_mediaPlayerPresenter.Opacity = 1;
+				}
+			});
+		}
+
+		#endregion
+
+		#region AreTransportControlsEnabled Property
+
+		public bool AreTransportControlsEnabled
+		{
+			get { return (bool)GetValue(AreTransportControlsEnabledProperty); }
+			set { SetValue(AreTransportControlsEnabledProperty, value); }
+		}
+
+		public static DependencyProperty AreTransportControlsEnabledProperty { get; } =
+			DependencyProperty.Register(
+				nameof(AreTransportControlsEnabled),
+				typeof(bool),
+				typeof(MediaPlayerElement),
+				new FrameworkPropertyMetadata(false));
+
+		#endregion
+
+		#region Stretch Property
+
+		public Stretch Stretch
+		{
+			get { return (Stretch)GetValue(StretchProperty); }
+			set { SetValue(StretchProperty, value); }
+		}
+
+		public static DependencyProperty StretchProperty { get; } =
+			DependencyProperty.Register(
+				nameof(Stretch),
+				typeof(Stretch),
+				typeof(MediaPlayerElement),
+				new FrameworkPropertyMetadata(Stretch.Uniform));
+
+		#endregion
+
+		public MediaTransportControls TransportControls { get; set; } = new MediaTransportControls();
+
+		public MediaPlayerElement() : base()
+		{
+		}
+
+		protected override void OnLoaded()
+		{
+			base.OnLoaded();
+
+			MediaPlayer.AutoPlay = AutoPlay;
+
+			if(MediaPlayer.PlaybackSession.PlaybackState == MediaPlaybackState.Opening && AutoPlay)
+			{
+				MediaPlayer.Play();
+			}
+		}
+		
+		protected override void OnApplyTemplate()
+		{
+			base.OnApplyTemplate();
+
+			_posterImage = this.GetTemplateChild(PosterImageName) as Image;
+			_mediaPlayerPresenter = this.GetTemplateChild(MediaPlayerPresenterName) as MediaPlayerPresenter;
+			_transportControlsPresenter = this.GetTemplateChild(TransportControlsPresenterName) as ContentPresenter;
+			_transportControlsPresenter.Content = TransportControls;
+			TransportControls.ApplyTemplate();
+
+			if (MediaPlayer == null)
+			{
+				MediaPlayer = new Windows.Media.Playback.MediaPlayer();
+			}
+
+			if (PosterSource != null && MediaPlayer.PlaybackSession.PlaybackState == MediaPlaybackState.None)
+			{
+				_posterImage.Visibility = Visibility.Visible;
+				_mediaPlayerPresenter.Opacity = 0;
+			}
+
+			if (!_isTransportControlsBound)
+			{
+				TransportControls?.SetMediaPlayer(MediaPlayer);
+				_isTransportControlsBound = true;
+			}
+		}
+
+		public void SetMediaPlayer(Windows.Media.Playback.MediaPlayer mediaPlayer)
+		{
+			MediaPlayer?.Dispose();
```

Comment:
maybe you should dispose from the DP callback instead of here

Reply:
Done